### ISSUE TITLE
7$: Error if const variable has no initializer

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#252]
quick-lint-js should report an error for the following code:
```js
const x;
```

**Comment by colin1227** [comment]
Can I try this one?

**Comment by strager** [comment]
@colin1227 Sure.

**Comment by colin1227** [comment]
@strager 
so if I understand, if a constant variable is only declared and not assigned a value, throw an error. right?

should there error message say this?:

 "{variableName}" is declared without an assigned initial value.

also, I am new to this repo. where do you want this logic?

**Comment by strager** [comment]
> @strager
> so if I understand, if a constant variable is only declared and not assigned a value, throw an error. right?

Correct.

> should there error message say this?:
> 
> "{variableName}" is declared without an assigned initial value.

Yes, that sounds good.

> also, I am new to this repo. where do you want this logic?

Probably in parser::parse_and_visit_let_bindings.

**Comment by Akhil132-dev** [comment]
its a rule in js for const that JavaScript const variables must be assigned a value when they are declared.

**Comment by Akhil132-dev** [comment]
can you help me to setup this project to my local machine so i can work on it

**Comment by strager** [comment]
> can you help me to setup this project to my local machine so i can work on it

Did you see the BUILDING documentation? https://github.com/quick-lint/quick-lint-js/blob/9bdb827a8e7e4728f53234fef1231aa0a747f14c/docs/BUILDING.md

FYI, @colin1227 claimed this task. (Are you still hacking on it, colin?)

**Comment by colin1227** [comment]
@strager yes, sorry, been busier than expected, I'll try to get something committed today

**Comment by colin1227** [comment]
WIP: 
- I still have loose comments
- 3 errors about scoping, string comparison, and something else
- need to write and run tests
[but some progress](https://github.com/colin1227/quick-lint-js/tree/const-no-initalizer-252)

**Comment by colin1227** [comment]
@strager will you be streaming this evening? I want to get a PR up but I need a better understanding of the function I am working in and would like to ask you some questions.

**Comment by strager** [comment]
> @strager will you be streaming this evening?

No. I will next stream Monday (about 49 hours from now).

**Comment by colin1227** [comment]
[new branch](https://github.com/colin1227/quick-lint-js/tree/const-252-new) based on what you mentioned about working within the `parse_and_visit_let_bindings` function instead of the linter.

1.) can't compile rn, current error: `error: array initializer must be an initializer list` line 3018 where I define a variable.
2.) I don't know for sure how to get the declaration token [but I hope I am on the right path](https://github.com/colin1227/quick-lint-js/blob/const-252-new/src/quick-lint-js/parse.h#L3010).
3.) where do I write tests? how many tests?

**Comment by colin1227** [comment]
@strager I need to get back to making websites that is where my programming skills excel. I appreciate you giving me a wack at this but I need to work within my comfort zone and start to actually make money coding again. maybe when I have more job security I can dump more time into c++ but not right now.

**Comment by singalhimanshu** [comment]
I don't think anyone is working on this task, @strager can I work on it?

**Comment by strager** [comment]
> I don't think anyone is working on this task, @strager can I work on it?

Yes.
